src/NuGet.Core/NuGet.Protocol/NuGet.Protocol.csproj

```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(TargetFrameworksLibraryForSigning)</TargetFrameworks>
+    <TargetFrameworks>$(TargetFrameworksLibraryForSigning);$(NETCoreTargetFramework)</TargetFrameworks>
```

Comment:
Adding a framework to one of our assemblies is a significant change, and one that had both me & Kartheek wondering what it's for. 
It's something that should be a part of the PR description. 


I think there's some side-effects we need to work through, some documented in https://github.com/NuGet/NuGet.Client/blob/dev/docs/updating-target-frameworks.md:

1. The previous frameworks were net472, ns2.0, net5.0 and now net8.0. 
The test frameworks are net472;net8.0;netcoreapp3.1

Each test framework will be running against the most compatible NuGet.Protocol framework. 
net472 tests net472, net8.0 tests net8.0, netcoreapp3.1 tests ns2.0. 
This leaves net5.0 untested. 

2. There's also the source build considerations. This change may lead to some duplicates in the source build work that *may* break them. 

Let's explore what options we have and make the best decision. 
Some potential ideas we can discuss (there's probably more)
- Say we won't test net5.0 from NuGet.Protocol
- Bump all the frameworks from net5.0 to net8.0 (breaking change for our packages, but I it could simplify our testing)
- make NuGet.Protocol target net8.0 only instead of net5.0

Reply:
Can we not add net5.0 to the list of TFMs of the test projects? That way we don't break customers who are on net 5.0

Reply:
We can do that too. 

We've been having a separate discussion with @zivkan, we have a lot of frameworks in our packages right now and that does make our support matrix larger. 

net5.0 as a runtime is out of support, so I'd expect that anyone having a runtime dependency on NuGet's libraries will likely be on a newer runtime. 

We can also say, if you really only need net5.0, use an older version of our packages. 

I think it'd be great to consolidate those conversations into 1.

Reply:
PR is out:
* https://github.com/NuGet/NuGet.Client/pull/6086

Let's converse over there about this

Reply:
I will rebase after this is merged